src/pages/index.js

```diff
@@ -121,3 +186,92 @@ const editProfileFormValidator = new FormValidator(
 );
 editProfileFormValidator.enableValidation();
 editProfileFormValidator.disableButton();
+
+/*******************
+ * New code before sorting*
+ ******************/
+
+function handleDeleteCard(cardData) {
+  cardData.cardElement.remove();
+  deleteCardModal.close();
+}
+
+const avatarEditModal = new PopupWithForm(
+  "#avatar-edit-modal",
+  handleAvatarEditSubmit
+);
+avatarEditModal.setEventListeners();
+
+function handleAvatarEditSubmit(data) {
+  avatarEditModal.setIsLoading(true);
+  api
+    .updateUserAvatar(data.avatar)
+    .then(() => {
+      userInfo.setUserInfo({ avatar: data.avatar });
+      avatarEditModal.close();
+    })
+    .catch((err) => console.error(`Error updating avatar: ${err}`))
+    .finally(() => avatarEditModal.setIsLoading(false));
+}
+
+const avatarEditForm = document.querySelector("#avatar-edit-form");
+const avatarFormValidator = new FormValidator(
+  validationSettings,
+  avatarEditForm
+);
+avatarFormValidator.enableValidation();
+
```

Comment:
Please, add `avatarFormValidator.disableButton();`  in line 223 to disable the `avatar` save button on load https://snipboard.io/RpTCkj.jpg